src/magic_folder/test/test_web.py

```diff
@@ -260,13 +281,22 @@ def treq_for_folders(basedir, auth_token, folders):
     :param folders: A mapping from Magic Folder names to their configurations.
         These are the folders which will appear to exist.
 
+    :param bool start_folder_services: If ``True``, start the Magic Folder
+        service objects.  Otherwise, don't.
+
     :return: An object like the ``treq`` module.
     """
     global_config = create_global_configuration(
         basedir,
-        u"non-endpoint:",
+        # Make this endpoint string and the one below parse but make them
+        # invalid, too, because we don't want anything to start listening on
+        # these during this set of tests.
+        u"tcp:-1",
+        # It wants to know where the Tahoe-LAFS node directory is but we don't
+        # have one and we don't want to invoke any functionality that requires
+        # one.  Give it something bogus.
         FilePath(u"/non-tahoe-directory"),
-        u"non-endpoint:",
+        u"tcp:-1",
```

Comment:
I don't know if this is viable, but can we register an endpoint plugin for testing? So then we could do "real endpoint stuff" (like listen) but avoid doing "real network stuff" too, maybe. I don't want to hold up this PR for that, just asking/noting it here after seeing the other comment below (about starting just some services).

Reply:
We can do that, yea, and it's a big strategy I used in Tahoe-LAFS to fix a lot of the "address already in use" errors.  It's a lot of work though, which is why I skipped it for now here and did this cheesy thing.

I'll leave a comment here explaining this situation, though, linked to a ticket.